### DISCUSSION TITLE
toggle problem

### DISCUSSION THREAD

**Discussion by jasha1974** [Power-Components/livewire-powergrid#181] [Q&A] [answered]
Hi, at first, nice component, but i've got problem with toggle, more at captured video. Same problem in your demo.

https://user-images.githubusercontent.com/38765749/145780774-63501577-35ec-4f77-aae1-98c299cc37df.mp4

thanmks a lot Slavik J.

**Comment by luanfreitasdev** [comment, marked as answer]
Maybe it has to do with the Alpine version. Soon we will be uploading the official version to Alpine 3

**Reply by jasha1974** [reply]
ok, i'll wait, thanks

**Reply by diegoba73** [reply]
Hi jasha, If you put defer in your js, the toggle starts working, the explanation is here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script
it would look something like this: "<script src="{{asset('js/app.js')}}" defer></script>"
I hope it helps you

**Reply by EmilioVar** [reply]
I am with the same problem. In theory Alpine 3 should be used but the same thing keeps happening

**Comment by EmilioVar** [comment]
It seems the problem is because turning the toggle on/off refreshes the table but remembers the previous state. Then, by wanting to give it again, it burdens us with the state prior to it. If we select any other element, we see how the one that was causing problems returns to its correct state.

I have managed to solve it by doing the following:

At the end of the onUpdatedToggleable method (the one that receives the toggle change request), I add the call to the following method:

$this->skipRender();

It is a Livewire method that allows you to skip the rendering of the component and, since powergrid is powered by livewire, we can use this method to avoid reloading the component and then confusing it with its state.

https://livewire.laravel.com/docs/actions#skipping-re-renders
